db/sql/deploy/buildings/functions/buildings_update_suburb_locality.sql

```diff
@@ -0,0 +1,141 @@
+        AND bsl.external_suburb_locality_id = ic.old_id
+        AND (ST_Within(ST_Expand(buildings.building_outlines.shape, 100), ic.shape)
+        OR (ST_Overlaps(ST_Expand(buildings.building_outlines.shape, 100), ic.shape)
+        AND ST_Area(ST_Intersection(buildings.building_outlines.shape, ic.shape)) >= buildings_reference.suburb_locality_area_of_second_largest_intersect(buildings.building_outlines.shape, ic.new_id)
+        ))
+        RETURNING *
+        )
+
+    SELECT count(*)::integer FROM building_outline_updates;
+
+$$
+LANGUAGE sql VOLATILE;
+
+COMMENT ON FUNCTION buildings.building_outlines_update_changed_and_deleted_suburb() IS
+'Replace suburb values with the intersection result of buildings in the building_outlines table';
+
+
+-- building_outlines_update_added_suburb (replace suburb values with the intersection result)
+    -- params: 
+    -- return: integer count of number of building outlines updated
+
+CREATE OR REPLACE FUNCTION buildings.building_outlines_update_added_suburb()
+RETURNS integer AS
+$$
+    -- add new suburbs to buildings_reference.suburb_locality
+      -- (to get new suburb_locality_ids)
+    WITH add_new_suburbs AS (
+      INSERT INTO buildings_reference.suburb_locality (
+        external_suburb_locality_id, suburb_4th,
+        suburb_3rd, suburb_2nd, suburb_1st, shape
+      )
+      SELECT
+        id, suburb_4th, suburb_3rd,
+        suburb_2nd, suburb_1st,
+        ST_Transform(shape, 2193)
+      FROM 
+        admin_bdys.nz_locality
+      WHERE
+        type IN ('ISLAND','LOCALITY','PARK_RESERVE','SUBURB')
+        AND id NOT IN (
+          SELECT
+            external_suburb_locality_id
+          FROM 
+            buildings_reference.suburb_locality
+        ) RETURNING *
+    ),
+
+    -- Find the 'old suburb locality ids' that overlap the new suburbs
+    added_overlap AS (
+      SELECT
+        ans.suburb_locality_id as new_suburb_loc_id,
+        bsl.suburb_locality_id as old_suburb_loc_id,
+        ans.shape AS shape
+      FROM
+        add_new_suburbs ans,
+        buildings_reference.suburb_locality bsl
+      WHERE 
+        (
+          ST_Overlaps(bsl.shape, ans.shape)
+          OR ST_Within(bsl.shape, ans.shape)
+        )
+        AND ans.suburb_locality_id != bsl.suburb_locality_id
+    ),
+
+    -- update building outlines where:
+      -- they have a id that is an old id (that overlaps a new suburb)
+      -- the building is within/overlaps the new suburb
+      -- the building overlaps the new polygon by the largest proportion 
+    update_building_outlines AS (
+        UPDATE 
+          buildings.building_outlines b
+        SET 
+          suburb_locality_id = ao.new_suburb_loc_id
+        FROM 
+          added_overlap ao
+        WHERE 
+          b.suburb_locality_id = ao.old_suburb_loc_id
+          AND (ST_Within(ST_Expand(b.shape,100), ao.shape)
+          OR (ST_Overlaps(ST_Expand(b.shape,100), ao.shape)
```

Comment:
We previously used ST_Expand 1000m.  Consider changing back to this or ensure all buildings are within 100m of LOCALITY/ISLAND/SUBURB/PARK_RESERVE